### DISCUSSION TITLE
写了一个openai的安全反代docker，提升部署安全

### DISCUSSION THREAD

**Discussion by nenew** [ChatGPTNextWeb/NextChat#1404] [Show and tell]
我写了一个docker，帮助部署一个安全的api反代。很欣赏你的作品，Good Job.
可以看下我的git：https://github.com/nenew/openai-api
docker在https://hub.docker.com/repository/docker/nenew/openai-api/general

**Comment by Yidadaa** [comment]
关于仓库中对于“滥用”部分的说明，openai 做风控的时候不可能不考虑开发者使用 severless 服务部署项目的需求，所以我认为 api 风控策略和所谓的 ip 不会强相关，还是会基于 api 所属账户的等级来判定。但是网页版由于每个有效账户都能免费访问，风控策略会更严格，会和 ip 相关。

**Reply by nenew** [reply]
我目前看到的用nginx的反代都会设置["X-Real-IP"]和["X-Forwarded-For"]，这两个就会暴露访问者的真实IP，比如把你的项目假设在国内，然后api用官网，如果api做个收集，可以直接对大面积的国内访问使用的key来封禁以及封号。还有一点就是针对IP的访问，如果同IP多次注册或者使用不同的key也会有影响，公开代理很多都无法注册了，这个就是根据IP来检测的。
你的作品很棒，我也在使用，虽然现在都还在排队等4.0的api，我可能更多是出于安全性考虑，防止封号。

**Reply by Yidadaa** [reply]
IP 多次注册封禁这个不亏，号贩子会破坏整个生态，和矿老板囤卡一个道理。至于你说的国区封禁，根据我现在掌握的信息，openai 基本是睁一只眼闭一只眼，只要你按时给他订阅送钱就行。
